src/modules/colorPicker/ColorPicker/dllmain.cpp

```diff
@@ -150,12 +232,46 @@ class ColorPicker : public PowertoyModuleIface
         if (m_enabled)
         {
             ResetEvent(send_telemetry_event);
+            ResetEvent(m_hInvokeEvent);
             TerminateProcess(m_hProcess, 1);
         }
 
         m_enabled = false;
     }
 
+    virtual bool on_hotkey(size_t hotkeyId) override
+    {
+        if (m_enabled)
+        {
+            if (!is_process_running())
+            {
+                launch_process();
+            }
+
+            SetEvent(m_hInvokeEvent);
+            return true;
+        }
+
+        return false;      
+    }
+
+    virtual size_t get_hotkeys(Hotkey* hotkeys, size_t buffer_size) override
+    {
+        if (m_hotkey.key)
+        {
+            if (hotkeys && buffer_size >= 1)
+            {
+                hotkeys[0] = m_hotkey;
+            }
```

Comment:
Should we return 0 when buffer_size is zero?

Reply:
The interface of the function says that it should return the number of hotkeys available, and fill up the buffer to the extent possible. The caller can then deduce the number of elements filled, if that's ever needed.

https://github.com/microsoft/PowerToys/blob/master/src/modules/interface/powertoy_module_interface.h#L74

Reply:
The function does not work as expected if `buffer_size` is zero. It returns 1 but it has to return 0. It is an edge case.

Reply:
That's by design. Please read the comment on the line of code I posted above, and also take a look at this function to understand why we need to return 1 when passing an empty buffer.

https://github.com/microsoft/PowerToys/blob/master/src/runner/powertoy_module.cpp#L55

Reply:
Looks like Windows API